ovos_local_backend/utils/geolocate.py

```diff
@@ -103,72 +102,96 @@ def _geolocate(self, address):
                 error = location_data["raw"]["error"]
             elif location_data.get("accuracy") == "Unmatchable":
                 error = "No results found"
+            else:
+                location = Location(location_data)
 
-            data["raw"] = location_data
-            data["country"] = location_data.get("country")
-            data["country_code"] = location_data.get("country_code")
-            data["region"] = location_data.get("region")
-            data["address"] = location_data.get("address")
-            data["state"] = location_data.get("state")
-            data["confidence"] = location_data.get("confidence")
-            data["lat"] = location_data.get("lat")
-            data["lon"] = location_data.get("lng")
-            data["city"] = location_data.get("city") or location_data.get("address")
-
-            data["postal"] = location_data.get("postal")
-            data["timezone"] = location_data.get("timezone_short")
-        if not data:
+        if not location:
             error = "No results found"
         if error:
             raise RuntimeError(error)
-        return data
+        return location
 
     def get_location(self, address):
 
         if self.provider == GeocoderProviders.SELENE:
             return self.engine(address)  # selene proxy, special handling
+        
+        return self._geolocate(address)
+
+
+class Location:
+    def __init__(self, location_data):
+        if not isinstance(location_data, dict):
+            raise TypeError
+        self._raw = location_data
+        # backwards comp
+        old_conf = isinstance(location_data.get("city"), dict)
+
+        self.city = location_data["city"]["name"] if old_conf else \
+                location_data.get("city") or location_data.get("address")
+        self.address = location_data.get("address")
```

Comment:
Same as above.  I use the "name" tag, and another tag "address" would be useful, just not a replacement

Reply:
i actually can't follow. `city` _is_ the `name` . The formerly used `code` was arbitrary and is not used by any geolocation service at this point.

Some geolocation services only send `address`, thus `location_data.get("city") or location_data.get("address")`

Reply:
"address" here is meant for users that send the full string and we geolocate from that, can be removed if needed but my thinking is that this would integrate with the backend manager (where this field can be entered)